src/bp/core/services/dialog/flow/service.ts

```diff
@@ -324,14 +327,19 @@ export class ScopedFlowService {
     const flowFiles = await this.ghost.directoryListing(FLOW_DIR, '**/*.json')
 
     const isNew = !flowFiles.find(x => flow.location === x)
+
+    // the onFlowChange function called inside prepareSaveFlow can cause
+    // cache events to be fired. We aren't interested in catching any of those
+    this.setExpectedSaves(flow.name, 999999)
     const { flowPath, uiPath, flowContent, uiContent } = await this.prepareSaveFlow(flow, isNew)
 
+    this.setExpectedSaves(flow.name, 2)
+    this.invalidateFlow(flow.name, flow)
```

Comment:
This feels like a hack... not sure it's a good idea. 99999 ? Magical number

Reply:
It's pretty hackish I agree but it solves a problem for now. A client really needs this so I put todos to remind us to come back to this when we improve the caching